sky/utils/command_runner.py

```diff
@@ -352,22 +355,21 @@ def rsync(
             ])
         command = ' '.join(rsync_command)
 
-        returncode, _, stderr = log_lib.run_with_log(command,
-                                                     log_path=log_path,
-                                                     stream_logs=stream_logs,
-                                                     shell=True,
-                                                     require_outputs=True)
-        while returncode != 0 and retry_cnt > 0:
+        backoff = common_utils.Backoff(initial_backoff=5, max_backoff_factor=5)
```

Comment:
max_retry += 1?

Otherwise, defaulting to 1 means no retry is performed.

Reply:
The loop checks for `max_retry >= 0`, so when it has the default value of 1, at most 1 retry will be made. (1st iteration `max_retry` is 1, second iteration `max_retry` is 0)